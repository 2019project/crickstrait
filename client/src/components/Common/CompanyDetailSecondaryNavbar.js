import React, { Component } from "react";
import {
  getCompanyDetailById,
  getVolatility
} from "../../actions/CompanyDetail";
import { connect } from "react-redux";
import companylogo from "./stockslogo.PNG";

// importing css file
import "../../styles/CompanyDetailSecondaryNavbar.css";
import { Link } from "react-router-dom";

export class CompanyDetailSecondaryNavbar extends Component {
  componentDidMount() {
    this.props.getVolatility("AAPL");
  }
  render() {
    return (
      <div>
        <div id="parent">
          {this.props.company ? (
            <div>
              {this.props.company.map(coms => (
                <>
                  <div id="company-details-tab-container">
                    <>
                      <div id="secondary-navbar-img-name-container">
                        <div id="secondary-navbar-img">
                          <img
                            id="stocks_img"
                            src={
                              coms.image == null
                                ? companylogo
                                : "data:image/jpeg;base64," + coms.image
                            }
                          />
                        </div>

                        <div id="secondary-navbar-ticker_name">
                          <p>{coms.ticker_name}</p>
                          <p id="CompanyDetailsIndexClose">
                            {coms.last_share_price}

                            <sub id="secondary-navbar-sub">USD</sub>
                          </p>
                          <p id="last_date">
                            CLOSED PRICE (
                            {new Date(coms.share_date).toLocaleDateString(
                              "en-IN",
                              {
                                month: "short",
                                day: "2-digit",
                                year: "numeric"
                              }
                            )}
                            )
                          </p>
                        </div>
                        <div />
                      </div>
                      <ul id="company-detail-content-container">
                        <Link
                          className="company-link"
                          // PASSING TO COMPANY DETAIL PAGE WITH THE ID WHICH IS MAPPED FROM THE REDUCER
                          to={{
                            pathname: "/companydetail/" + coms.id
                          }}
                        >
                          <span
                            id="company-detail-overview-click"
                            // onClick={() => {
                            //   // changing the values of the state while clicking, the clicked component will be set to true others will be false
                            //   // this.setState({
                            //   //   overview: true,
                            //   //   financial: false,
                            //   //   analysis: false,
                            //   // });
                            //   console.log("Overview clicked");
                            // }}
                          >
                            <li
                              id="comapany-detail-overview-li"
                              // different css properties based on the value of the state
                              className={
                                this.props.selected === "overview"
                                  ? "options-selected-li"
                                  : "options-li"
                              }
                            >
                              Overview
                            </li>
                          </span>
                        </Link>
                        {/* Financials  */}
                        {/* <div id="company-detail-financial"> */}

                        <div id="company-detail-financial">
                          <Link
                            className="company-link"
                            to={{
                              pathname: "/financial/" + coms.id
                            }}
                          >
                            <span
                              id="company-financial-click"
                              // onClick={() => {
                              //   this.setState({
                              //     overview: false,
                              //     financial: true,
                              //     analysis: false,
                              //   });
                              // }}
                            >
                              <li
                                id="company-detail-financial-li"
                                className={
                                  this.props.selected === "financial"
                                    ? "options-selected-li"
                                    : "options-li"
                                }
                              >
                                Financials
                              </li>
                            </span>
                          </Link>
                        </div>

                        <div id="company-detail-analysis">
                          <Link
                            className="company-link"
                            to={{
                              pathname: "/analysis/" + coms.id
                            }}
                          >
                            <span
                              id="company-analysis-click"
                              // onClick={() => {
                              //   this.setState({
                              //     overview: false,
                              //     financial: false,
                              //     analysis: true
                              //   });
                              // }}
                            >
                              <li
                                id="company-detail-analysis-li"
                                className={
                                  this.props.selected === "analysis"
                                    ? "options-selected-li"
                                    : "options-li"
                                }
                              >
                                Analysis
                              </li>
                            </span>
                          </Link>
                        </div>
                      </ul>
                      <div id="secondary-nav-values">
                        <div id="voltality">
                          <p> {this.props.voltality} </p>
                          <div id="voltality_tooltip">
                            <div id="kk">VOLATILITY </div>
                            <div id="abc">
                              <i class="fa fa-question-circle" />
                              <span id="volatility">
                                Volatility is a statistical measure of the
                                dispersion of returns for a given security or
                                market index. In most cases, the higher the
                                volatility, the riskier the security. Volatility
                                is often measured as either the standard
                                deviation or variance between returns from that
                                same security or market index.
                              </span>
                            </div>
                          </div>
                        </div>
                        {/* ------------------------------------------------------------------ */}
                        <div id="voltality">
                          <p id="market_cap"> {coms.last_market_cap} </p>
                          <div id="kk">MARKET CAP</div>
                        </div>
                        {/* --------------------------------------------------------------------------- */}
                      </div>
                    </>

                    <div id="bla">
                      {localStorage.getItem("token") ? (
                        <div id="secondary-navbar-bla">
                          {/* second grid of secondaru navbar */}
                          <button
                            id="downloadButton"
                            onClick={async () => {
                              fetch(
                                "http://localhost:2001/api/companydetail/indicatorsdata/" +
                                  coms.ticker_name
                              ).then(response => {
                                response.blob().then(blob => {
                                  let url = window.URL.createObjectURL(blob);
                                  let a = document.createElement("a");
                                  a.href = url;
                                  a.download = coms.ticker_name + ".csv";
                                  a.click();
                                });
                                //window.location.href = response.url;
                              });
                              // this.props.downloadOHLC("AAPL");
                            }}
                          >
                            <i class="fa fa-download" /> Download
                          </button>
                        </div>
                      ) : (
                        <div style={{ display: "none" }}></div>
                      )}
                    </div>
                  </div>
                  <></>
                </>
              ))}
            </div>
          ) : (
            <p />
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  company: state.CompanyDetailReducer.company,
  voltality: state.CompanyDetailReducer.voltality
});
export default connect(mapStateToProps, {
  getCompanyDetailById,
  getVolatility
})(CompanyDetailSecondaryNavbar);
