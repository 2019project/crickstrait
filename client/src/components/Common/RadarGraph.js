import React, { Component } from "react";
import { Polar } from "react-chartjs-2";
import { connect } from "react-redux";

export class RadarGraph extends Component {
  state = {
    marksData: {
      labels: [
        "Market Cap",
        "Net Profit",
        "Share Price",
        "Revenue",
        "Total Assets",
        "EBIT"
      ],
      datasets: [
        {
          fill: true,
          backgroundColor: [
            "#FF6384",
            "#4BC0C0",
            "#FFCE56",
            "#E7E9ED",
            "#36A2EB",
            "#2c3e50"
          ],

          data: this.props.data
        }
      ]
    }
  };
  render() {
    return (
      <div>
        <Polar
          options={{
            legend: false,
            labels: false,
            responsive: true
          }}
          data={this.state.marksData}
        />
      </div>
    );
  }
}
const mapStateToProps = state => ({});
export default connect(mapStateToProps, {})(RadarGraph);
