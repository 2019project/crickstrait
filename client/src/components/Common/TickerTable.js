import React, { Component } from "react";
import "../../styles/TickerTable.css";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

export class TickerTable extends Component {
  render() {
    console.log(this.props.isIndex);
    console.log(this.props.tableHeaders);
    return (
      <div className="table__container">
        <table className="ticker__table">
          <thead className="ticker__tableHeader">
            <tr>
              {this.props.tableHeaders.map((data, index) => (
                <th>{data}</th>
              ))}
            </tr>
          </thead>
          {console.log("tabledata", typeof this.props.tableData)}
          {this.props.tableData.map((current, index) => (
            <tr>
              {this.props.isIndex == true ? (
                <Link
                  to={{ pathname: "/indexProfile/" + current.ticker_id }}
                  style={{ textDecoration: "none" }}
                >
                  <td>{current.ticker_name ? current.ticker_name : null}</td>
                </Link>
              ) : (
                <Link
                  to={{ pathname: "/companydetail/" + current.ticker_id }}
                  style={{ textDecoration: "none" }}
                >
                  <td>{current.ticker_name ? current.ticker_name : null}</td>
                </Link>
              )}
              {current.tickerValues
                ? Object.keys(current.tickerValues).map((keyName, index) => (
                    <>
                      {current.tickerValues[keyName].substring(0, 1) == "-" ? (
                        <td style={{ color: "#ff4d4d" }}>
                          {current.tickerValues[keyName]}
                        </td>
                      ) : current.tickerValues[keyName].substring(0, 1) ==
                        "+" ? (
                        <td style={{ color: "#27ae60" }}>
                          {current.tickerValues[keyName]}
                        </td>
                      ) : (
                        <td>{current.tickerValues[keyName]}</td>
                      )}
                    </>
                  ))
                : null}
            </tr>
          ))}
        </table>
      </div>
    );
  }
}
const mapStateToProps = state => ({});
export default connect(mapStateToProps, {})(TickerTable);
