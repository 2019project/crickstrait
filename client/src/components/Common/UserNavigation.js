import React, { Component } from "react";
import "../../styles/UserNavigation.css";
import wallet from "./wallet.png";
import { Link } from "react-router-dom";
import jwt_decode from "jwt-decode";

export default class UserProfile extends Component {
  state = {
    buyStocks: true,
    myStocks: false,
    historyStocks: false
  };

  render() {
    var decode = jwt_decode(localStorage.getItem("token"));
    return (
      <div>
        <div id="userNavigationContainer">
          <div id="userNavigation">
            <h1>Hi Nikhil !</h1>
            <img src={wallet} id="wallet" />{" "}
            <span id="walletPrice">
              $
              {String(this.props.wallet).length != 0
                ? this.props.wallet
                : decode.wallet}
            </span>
            <div id="userNavigationButtonContainer">
              <Link to="/profile">
                <button
                  id={this.state.buyStocks ? "buyStocks-active" : "buyStocks"}
                  onClick={() =>
                    this.setState({
                      buyStocks: true,
                      myStocks: false,
                      historyStocks: false
                    })
                  }
                >
                  Buy Stocks
                </button>
              </Link>
              <Link to="/buyStocks">
                <button
                  id={this.state.myStocks ? "myStocks-active" : "myStocks"}
                  onClick={() =>
                    this.setState({
                      buyStocks: false,
                      myStocks: true,
                      historyStocks: false
                    })
                  }
                >
                  My Stocks
                </button>
              </Link>
              <Link to="/userhistory">
                <button
                  id={
                    this.state.historyStocks
                      ? "historyStocks-active"
                      : "historyStocks"
                  }
                  onClick={() =>
                    this.setState({
                      buyStocks: false,
                      myStocks: false,
                      historyStocks: true
                    })
                  }
                >
                  History
                </button>
              </Link>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
