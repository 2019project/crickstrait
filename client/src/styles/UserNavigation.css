@import url(https://fonts.googleapis.com/css?family=Karla);
* {
  font-family: Karla;
}
#userNavigationContainer {
  width: 100%;
  display: grid;
  grid-template-rows: 150px;
  text-align: center;
}

#userNavigation {
  color: #2c3e50;
}
#userNavigation h1 {
  font-weight: 700;
  font-family: Karla;
}
#wallet {
  width: 60px;
  height: 60px;
  margin-bottom: 20px;
}

#walletPrice {
  font-size: 25px;
  margin-top: 10px;
}

#userNavigationButtonContainer {
  display: flex;
  font-size: 18px;
}

#buyStocks {
  width: 100%;
  padding: 20px;
  font-family: Karla;
  cursor: pointer;
  color: #707070;
  background-color: #fff;
  outline: none;
  border: none;
  border-bottom: 1px solid #39abf7;
  border-top: 1px solid #39abf7;
}

#buyStocks-active {
  cursor: pointer;
  width: 100%;
  padding: 20px;
  font-family: Karla;
  color: #fff;
  background-color: #39abf7;
  outline: none;
  border: none;
}

#myStocks {
  width: 100%;
  padding: 20px;
  cursor: pointer;
  color: #707070;
  font-family: Karla;
  background-color: #fff;
  outline: none;
  border: none;
  border-bottom: 1px solid #39abf7;
}

#myStocks-active {
  cursor: pointer;
  width: 100%;
  padding: 20px;
  font-family: Karla;
  color: #fff;
  background-color: #39abf7;
  outline: none;
  border: none;
}

#historyStocks {
  width: 100%;
  padding: 20px;
  border-top: 1px solid #39abf7;
  cursor: pointer;
  font-family: Karla;
  color: #707070;
  background-color: #fff;
  outline: none;
  border: none;
  border-bottom: 1px solid #39abf7;
}

#historyStocks-active {
  cursor: pointer;
  width: 100%;
  font-family: Karla;
  padding: 20px;
  color: #fff;
  background-color: #39abf7;
  outline: none;
  border: none;
}

@media screen and (min-width: 430px) {
  #userNavigationContainer {
    width: 100%;
    grid-template-columns: 250px;
  }

  #userNavigationButtonContainer {
    display: block;
    font-size: 15px;
  }
}
