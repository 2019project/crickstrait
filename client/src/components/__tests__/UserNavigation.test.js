import React from "react";
import { shallow, mount } from "enzyme";
import { UserNavigation } from "../Common/UserNavigation";

const userNavigation = jest.fn();

const wrapper = shallow(
  <UserNavigation userNavigation={userNavigation} name={name} />
);

describe("Testing User  Secondary navbar Component", () => {
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("checks for on Buy Srtocks to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#buyStocks-active").simulate("click", e);
  });

  it("checks for on Buy Srtocks to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#buyStocks-active").simulate("click", e);
  });

  it("checks for on My Srtocks to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#myStocks").simulate("click", e);
  });

  it("checks for on History Srtocks to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#historyStocks").simulate("click", e);
  });
});
