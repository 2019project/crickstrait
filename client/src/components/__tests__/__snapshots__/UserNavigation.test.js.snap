// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User  Secondary navbar Component should mount the component 1`] = `
<div>
  <div
    id="userNavigationContainer"
  >
    <div
      id="userNavigation"
    >
      <h1>
        Hi User !
      </h1>
      <img
        id="wallet"
        src="wallet.png"
      />
       
      <span
        id="walletPrice"
      >
        $
        NaN
      </span>
      <div
        id="userNavigationButtonContainer"
      >
        <Link
          to="/profile"
        >
          <button
            id="buyStocks"
          >
            Buy Stocks
          </button>
        </Link>
        <Link
          to="/buyStocks"
        >
          <button
            id="myStocks"
          >
            My Stocks
          </button>
        </Link>
        <Link
          to="/userhistory"
        >
          <button
            id="historyStocks"
          >
            History
          </button>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
